### PR TITLE
PR: Require `applaunchservices` 0.3.0+

### LINKED ISSUE #17835: Problems with Spyder on Mac with Conda-forge and `applaunchservices`
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
We want to release a plugin to spyder on conda-forge. 
Our package must be a `noarch` type with the twist that python >= 3.8.
conda-forge insists  and we run `pip check` in the recipe (which makes a lot of sense), however this fails because spyder relays on `applaunchservices` for OSX, but not all it's dependencies are in conda-forge :frowning_face: and thus the check fails as follows: 

image
 
Fortunately for us, @goanpeca is the conda-forge reviewer, so I asked him if he could approve our feedstock when I do the following in or recipe: 

image

This will get me up and running, however the cleaner way would be if we take the `applaunchservices` out of spyder's conda-forge recipe, this way others (like this guy from `PROCITEC` ?) will not have the same problem as us.

All plugins can then enable the normal `pip check` without selectors (and discussions with conda-forge) and once the `applauchservices` work again, then we can enable it again and support for spyder on OSX under CONDA, the plugin recipies don't have to do anything anymore (presumed that the plugins are `noarch`)

Could this be done in the upcomming 5.3.1 release already ?

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- This version has the correct Pyobjc dependencies it relies on. Previous versions didn't, which generated problems in Conda-forge.
- This required to break our Conda requirements file into several files per operating system. I also decided to use environment yaml files for that because they have more metadata (e.g. the Anaconda channel we want to use).

### Issue(s) Resolved

Fixes #17835 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @ccordoba12

### CODE REVIEW

File: binder/environment.yml [hunk 1]
```diff
@@ -3,10 +3,9 @@ channels:
 dependencies:
 # We can not refer to an environment.yml file from another
 # So to get performant launches on mybinder.org, we have copied
-# the contents of this requirements/conda.txt to this file.yml.
+# the contents of this requirements/main.yml to this file.yml.
```
Comment (dalthviz) [inline binder/environment.yml:6]: Should this comment include the `linux.yml` file too?

**Review by dalthviz — DISMISSED** [review]
Thanks @ccordoba12  Left a comment about the comment in the binder env file. Other than that this LGTM :+1:

**Review by dalthviz — COMMENTED** [review]
I approved but it actually was a comment review 😅

File: binder/environment.yml (hunk 1 above)
Comment (ccordoba12) [reply binder/environment.yml]: Fixed in my last commit.

**Comment by ccordoba12** [comment]
> I approved but it actually was a comment review :sweat_smile:

Hehe, no problem.

**Review by dalthviz — APPROVED** [review]

**Pull request merged**